### DISCUSSION TITLE
strange behavior when table meets CJK character

### DISCUSSION THREAD

**Discussion by chenxuuu** [pradyunsg/furo#266] [Potential Issue] [answered]
sometimes the last character always in a new line alone, like these:

![QQ截图20210926183639](https://user-images.githubusercontent.com/10357394/134804441-a7d0d435-3e97-4714-9d66-065a0ec7af36.png)

![QQ截图20210926184110](https://user-images.githubusercontent.com/10357394/134804445-5aefe891-ed4d-4b6f-ae60-4d678af42ff4.png)

it should be this:

![QQ截图20210926183644](https://user-images.githubusercontent.com/10357394/134804493-fa2f87c1-fb10-4251-813d-73d986bc14b0.png)

test rst

```rst
========== ========
返回值类型 解释
========== ========
nil        无返回值
========== ========
```

**Comment by pradyunsg** [comment, marked as answer]
Hiya! Thanks for flagging this!

This seems to come from Sphinx itself, specifically from docutils (which is what Sphinx is built upon, to render the `.rst` files). Here's the final HTML that is provided to the Sphinx themes (in this case, Furo) to stylise:

```html
<table class="docutils align-default">
<colgroup>
<col style="width: 56%">
<col style="width: 44%">
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>返回值类型</p></th>
<th class="head"><p>解释</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>nil</p></td>
<td><p>无返回值</p></td>
</tr>
</tbody>
</table>
```

The issue is the colgroup, which seems to be have been calculated incorrectly for CJK characters, and forces that rewrapping. I'm able to reproduce the issue with a plain `.html` file as well:

```html
<table class="docutils align-default" border="1">
<colgroup>
<col style="width: 56%">
<col style="width: 44%">
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>返回值类型</p></th>
<th class="head"><p>解释</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>nil</p></td>
<td><p>无返回值</p></td>
</tr>
</tbody>
</table>
```

Dropping the colgroup fixes things, so does tweaking the percentages on the colgroup to `55-45`.

---

Overall, this is definitely a bug. It originates from a part of the toolchain that I don't control though, and I'd encourage you to reach out to the docutils project (see "If you find a bug" in https://docutils.sourceforge.io/#overview) and pursue them to fix this.

**Reply by chenxuuu** [reply]
thanks for your detailed explanation! 

my interim solution
```javascript
window.onload = function () {
    $("col").css("width","");
}
```
